tests/Versioning/VersionManagerTest.php

```diff
@@ -420,6 +449,54 @@ public function testRestoreWithNewChildren()
         $this->assertFalse($node->hasNode('newChildNode'));
     }
 
+    public function testRestoreWithChildrenInitialize()
+    {
+        $nodeTypeManager = $this->session->getWorkspace()->getNodeTypeManager();
+
+        $wildcardPropertyDefinition = $nodeTypeManager->createPropertyDefinitionTemplate();
+        $wildcardPropertyDefinition->setName('*');
+
+        $testNodeType = $nodeTypeManager->createNodeTypeTemplate();
+        $testNodeType->setName('nt:test-initialize-children');
+        $testNodeType->getPropertyDefinitionTemplates()->append($wildcardPropertyDefinition);
+
+        $testChildNodeType = $nodeTypeManager->createNodeTypeTemplate();
+        $testChildNodeType->setName('nt:child');
+        $testChildNodeType->getPropertyDefinitionTemplates()->append($wildcardPropertyDefinition);
+
+        $childNodeDefinition = $nodeTypeManager->createNodeDefinitionTemplate();
+        $childNodeDefinition->setName('child');
+        $childNodeDefinition->setDefaultPrimaryTypeName('nt:child');
+        $childNodeDefinition->setRequiredPrimaryTypeNames(['nt:child']);
+        $childNodeDefinition->setAutoCreated(true);
+        $childNodeDefinition->setMandatory(true);
+        $childNodeDefinition->setOnParentVersion(OnParentVersionAction::INITIALIZE);
+        $testNodeType->getNodeDefinitionTemplates()->append($childNodeDefinition);
+
+        $nodeTypeManager->registerNodeType($testChildNodeType, true);
+        $nodeTypeManager->registerNodeType($testNodeType, true);
```

Comment:
Are there other tests which perform the same actions that you perform here? Could we be uncovering an as-yet unknown bug in Jackalope Jackrabbit? Guess you already tried using something other than `nt:child` as a name?